ui/src/app/settings/selectors/config.js

```diff
@@ -0,0 +1,119 @@
+import { createSelector } from "reselect";
+
+const config = {};
+
+/**
+ * Returns list of all MAAS configs
+ * @param {Object} state - The redux state.
+ * @returns {Array} A list of all state.config.items.
+ */
+config.all = state => state.config.items;
+
+/**
+ * Returns true if users are loading.
+ * @param {Object} state - The redux state.
+ * @returns {Boolean} User is loading.
+ */
+config.loading = state => state.config.loading;
+
+/**
+ * Returns true if users have been loaded.
+ * @param {Object} state - The redux state.
+ * @returns {Boolean} User has loaded.
+ */
+config.loaded = state => state.config.loaded;
+
+/**
+ * Returns the MAAS config for default storage layout.
+ * @param {Object} state - The redux state.
+ * @returns {String} Default storage layout.
+ */
+config.defaultStorageLayout = createSelector(
+  [config.all],
+  configs => {
+    const configObj = configs.find(
+      config => config.name === "default_storage_layout"
+    );
+    if (configObj) {
+      return configObj.value;
+    }
+    return;
+  }
+);
+
+/**
+ * Returns the possible storage layout options reformatted as objects.
+ * @param {Object} state - The redux state.
+ * @returns {Array} Storage layout options.
+ */
+config.storageLayoutOptions = createSelector(
+  [config.all],
+  configs => {
+    const configObj = configs.find(
+      config => config.name === "default_storage_layout"
+    );
+    if (configObj) {
+      return configObj.choices.map(choice => ({
+        value: choice[0],
+        label: choice[1]
+      }));
+    }
+    return;
+  }
+);
+
+/**
+ * Returns the MAAS config for enabling disk erase on release.
+ * @param {Object} state - The redux state.
+ * @returns {Boolean} Enable disk erasing on release.
+ */
+config.enableDiskErasing = createSelector(
+  [config.all],
+  configs => {
+    const configObj = configs.find(
+      config => config.name === "enable_disk_erasing_on_release"
+    );
+    if (configObj) {
+      return configObj.value;
+    }
+    return;
+  }
+);
+
+/**
+ * Returns the MAAS config for enabling disk erase with secure erase.
+ * @param {Object} state - The redux state.
+ * @returns {Boolean} Enable disk erasing with secure erase.
+ */
+config.diskEraseWithSecure = createSelector(
+  [config.all],
+  configs => {
+    const configObj = configs.find(
```

Comment:
As this pattern is repeated several times, could we factor this out to a helper function with takes `configs` and the config name?